sympy/matrices/decompositions.py

```diff
@@ -1144,23 +1174,129 @@ def _QRdecomposition(M):
     [14,  21, -14],
     [ 0, 175, -70],
     [ 0,   0,  35]])
+
+    If the matrix is square and full rank, the $Q$ matrix becomes
+    orthogonal in both directions, and needs no augmentation.
+
+    >>> Q * Q.H
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+    >>> Q.H * Q
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+
     >>> A == Q*R
     True
 
-    QR factorization of an identity matrix:
+    A rank deficient matrix example:
 
-    >>> A = Matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    >>> A = Matrix([[12, -51, 0], [6, 167, 0], [-4, 24, 0]])
     >>> Q, R = A.QRdecomposition()
     >>> Q
     Matrix([
+    [ 6/7, -69/175],
+    [ 3/7, 158/175],
+    [-2/7,    6/35]])
+    >>> R
+    Matrix([
+    [14,  21, 0],
+    [ 0, 175, 0]])
+
+    It may return a $Q$ matrix which is orthogonal in only one way
+    around ($\mathbb{I} = Q^H Q$).
```

Comment:
Maybe this could be: "QRdecomposition might return a matrix Q that is rectangular. In this case the orthogonality condition might be satisfied as `I = Q.H*Q` but *not* in the reversed product `I = Q * Q.H`.